### PR TITLE
read only propagated downwards

### PR DESCRIPTION
**Pull request by jbbjarnason**

### CODE REVIEW

**Review by SindriTh — APPROVED** [review]

**Review by omarhogni — APPROVED** [review]
Is this sufficient? 

you cannot have f.e. read only std::maps using this

**Comment by jbbjarnason** [comment]
> Is this sufficient?
> 
> you cannot have f.e. read only std::maps using this

you are correct, will do those additions (also applies to vectors/array)

File: libs/stx/inc/public/tfc/utils/json_schema.hpp [hunk 1]
```diff
@@ -437,6 +439,9 @@ struct to_json_schema<T> {
       using V = std::decay_t<std::variant_alternative_t<I, T>>;
       auto& schema_val = (*s.oneOf)[I.value];
       schema_val.attributes.title = glz::name_v<V, true>;  // please MAKE sure you declare name within variants
+      if (s.attributes.read_only.has_value()) {
+        schema_val.attributes.read_only = s.attributes.read_only;
+      }
```
Comment (omarhogni) [inline libs/stx/inc/public/tfc/utils/json_schema.hpp:444]:
```suggestion
schema_val.attributes.read_only = s.attributes.read_only;
```
```suggestion
      schema_val.attributes.read_only = s.attributes.read_only;
```

**Review by omarhogni — APPROVED** [review]
It would be good to see some trivial tests for this. could encode a json schema and parse it back as normal json. Then the fields could be checked.

**Comment by jbbjarnason** [comment]
let's close for now this is a bit trickier than I thought

**Pull request closed without merging**